include/treelite/tree.h

```diff
@@ -347,7 +347,7 @@ class Tree {
    * \param nid ID of node being queried
    */
   inline std::vector<LeafOutputType> LeafVector(int nid) const {
-    if (nid > leaf_vector_offset_.Size()) {
+    if (static_cast<size_t>(nid) > leaf_vector_offset_.Size()) {
```

Comment:
I'll ask this question once, though I think it applies to most of the other places we're casting to `size_t`. Would it make more sense to have `nid` be a `size_t` to begin with? It's an array index which really should be a `size_t`, though I know that's a "purist" take that doesn't always play out in practice. Do we have a good reason not to make it a `size_t`?

Reply:
Let me try making it size_t.

Reply:
@wphicks It turns out it won't be wise to switch to `size_t`. Each tree node contains two integers to refer to the left and right child nodes, and switching to `size_t` will unnecessarily bloat the size of the model. Besides, we do not support trees with more than 2^31 nodes.